src/utils/get-youtube-video-id.js

```diff
@@ -0,0 +1,12 @@
+module.exports = function (url) {
+	return `<div data-responsive-youtube--container>
+    <iframe src="//www.youtube.com/embed/${getVideoIdFromURL(url)}" frameborder="0" allowfullscreen></iframe>
+  </div>`;
+};
+
+function getVideoIdFromURL(url) {
```

Comment:
I think JSDoc is less necessary here, but let's still add some for the sake of completion and expressiveness.

```suggestion
/**
 * Extracts a YouTube video's ID from its upload URL.
 * @param {string} url URL of the YouTube upload
 * @returns {string} the YouTube video's ID
 */
function getVideoIdFromURL(url) {
```

Reply:
this brings up an interesting philosophical discussion

i tend not to only add types and documentation to public/exported functions (in JavaScript).
my reasoning is that public methods are the only things that i guarantee a contract for private/support functions should be as disposable and simple to manipulate or tear down — provided the public contract remains the same.
in effect, the private/support functions are tested *thru* the public API

but i don't feel strongly about it either way, for this particular domain/scope

Reply:
I definitely agree with documenting exported/public functions. The moment you can't actually see the function you're using, the tooling becomes so much more crucial.

When in doubt on whether to add JSDoc to a utility function, I tend to lean towards yes, even though in this case, the only consumer is one other function in this same file. I opt for yes because it still helps get good tooling - since we indicate that `url` is a `string`, intellisense will autocomplete string methods, which can help us down the road if we needed to update or extend this function in any way.